js/ui/payg.js

```diff
@@ -612,6 +615,271 @@ var ApplyCodeNotification = GObject.registerClass({
+ * ┃ ││     └─────────────────────────────────────────────────────┘         ││ ┃
+ * ┃ │└─────────────────────────────────────────────────────────────────────┘│ ┃
+ * ┃ └───────────────────────────────────────────────────────────────────────┘ ┃
+ * ┃ ModalDialog.buttonLayout                                                  ┃
+ * ┃ ┌───────────────────────────────────┬───────────────────────────────────┐ ┃
+ * ┃ │                                   │                                   │ ┃
+ * ┃ │              Cancel               │            Add Credit             │ ┃
+ * ┃ │                                   │                                   │ ┃
+ * ┃ └───────────────────────────────────┴───────────────────────────────────┘ ┃
+ * ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+ */
+var PaygAddCreditDialog = GObject.registerClass(
+class PaygAddCreditDialog extends ModalDialog.ModalDialog {
+    _init() {
+        /* We want to be able to open the dialog multiple times per session
+         * without making the caller instatiate a new object before every call,
+         * so we need to disable destroyOnClose */
+        super._init({ styleClass: 'payg-add-credit-dialog',
+                      destroyOnClose: false });
+        super.connect('closed', this._onClosed.bind(this));
+
+        this.verificationStatus = UnlockStatus.NOT_VERIFYING;
+
+        /* This layout contains the prompt presented to the user and the labels
+         * reporting results to the user */
+        const title = _('Enter your unlock code');
+        let codeLength = Main.paygManager.codeLength;
+        let description = Gettext.ngettext(
+            'Enter a new keycode (%s character) to extend the time before your credit expires.',
+            'Enter a new keycode (%s characters) to extend the time before your credit expires.',
+            codeLength).format(codeLength);
+        this._promptLayout = new Dialog.MessageDialogContent({ title, description });
+
+        this._errorMessageLabel = new St.Label({
+            style_class: 'prompt-dialog-error-label',
+            visible: false,
+        });
+        this._errorMessageLabel.clutter_text.ellipsize = Pango.EllipsizeMode.NONE;
+        this._errorMessageLabel.clutter_text.line_wrap = true;
+        this._promptLayout.add_child(this._errorMessageLabel);
+
+        this._infoMessageLabel = new St.Label({
+            style_class: 'prompt-dialog-info-label',
+            visible: false,
+        });
+        this._infoMessageLabel.clutter_text.ellipsize = Pango.EllipsizeMode.NONE;
+        this._infoMessageLabel.clutter_text.line_wrap = true;
+        this._promptLayout.add_child(this._infoMessageLabel);
+
+        this.contentLayout.add_child(this._promptLayout);
+
+        /* This layout contains the code prefix, entry field and suffix */
+        this._codeEntryLayout = new St.BoxLayout({
+            x_expand: false,
+        });
```

Comment:
These days, no-property or single property constructors are written in a single line:

```javascript
this._codeEntryLayout = new St.BoxLayout({x_expand: false});
```